activerecord/lib/active_record/internal_metadata.rb

```diff
@@ -26,18 +26,30 @@ def table_exists?
         connection.table_exists?(table_name)
       end
 
-      # Creates an internal metadata table with columns +key+ and +value+
-      def create_table
-        unless table_exists?
-          key_options = connection.internal_string_options_for_primary_key
-
-          connection.create_table(table_name, id: false) do |t|
-            t.string :key, key_options
-            t.string :value
-            t.timestamps
+      def initialize_table
+        create_table
+        connection.reset!
+        record_current_environment
+      end
+
+      def record_current_environment
```

Comment:
Put public because https://github.com/rails/rails/pull/28046/files#diff-8d3c44120f7b67ff79e2fbe6a40d0ad6R1195 use this.